### PR TITLE
Staking message verification and processing (part I)

### PR DESCRIPTION
**Pull request by chaosma**
- add header V3 and body V2 for staking transactions
- add rawdb database read/write methods

### CODE REVIEW

File: core/types/block.go [hunk 1]
```diff
@@ -239,11 +244,21 @@ type extblockV1 struct {
 	IncomingReceipts CXReceiptsProofs
 }
 
+// includes staking transaction
+type extblockV2 struct {
+	Header           *block.Header
+	Txs              []*Transaction
+	Stks             []*staking.StakingTransaction
+	Uncles           []*block.Header
```
Comment (rlan35) [inline core/types/block.go]: can you remove Uncles. we don't need it.

File: internal/params/config.go [hunk 2]
```diff
@@ -137,6 +146,12 @@ func (c *ChainConfig) IsCrossTx(epoch *big.Int) bool {
 	return isForked(crossTxEpoch, epoch)
 }
 
+// IsStaking determines whether it is staking epoch
+func (c *ChainConfig) IsStaking(epoch *big.Int) bool {
+	stkEpoch := new(big.Int).Add(c.StakingEpoch, common.Big1)
```
Comment (rlan35) [inline internal/params/config.go]: no need to add 1. see the reasoning for IsCrossTx.

File: staking/types/messages.go [hunk 3]
```diff
@@ -49,10 +50,10 @@ func (d Directive) String() string {
 type NewValidator struct {
 	Description       `json:"description" yaml:"description"`
 	CommissionRates   `json:"commission" yaml:"commission"`
-	MinSelfDelegation *big.Int           `json:"min_self_delegation" yaml:"min_self_delegation"`
-	StakingAddress    common.Address     `json:"staking_address" yaml:"staking_address"`
-	PubKey            shard.BlsPublicKey `json:"validating_pub_key" yaml:"validating_pub_key"`
-	Amount            *big.Int           `json:"amount" yaml:"amount"`
+	MinSelfDelegation *big.Int       `json:"min_self_delegation" yaml:"min_self_delegation"`
+	StakingAddress    common.Address `json:"staking_address" yaml:"staking_address"`
+	PubKey            bls.PublicKey  `json:"validating_pub_key" yaml:"validating_pub_key"`
```
Comment (rlan35) [inline staking/types/messages.go]: don't change this. we always use 48 bytes for pubkey, don't use the struct object.

File: node/worker/worker.go [hunk 4]
```diff
@@ -145,63 +144,6 @@ func (w *Worker) SelectTransactionsForNewBlock(newBlockNum uint64, txs types.Tra
 	return selected, unselected, invalid
 }
 
-// SelectStakingTransactionsForNewBlock selects staking transactions for new block.
-func (w *Worker) SelectStakingTransactionsForNewBlock(
```
Comment (rlan35) [inline node/worker/worker.go]: use transaction flow for staking transaction too. So don't remove these.

File: core/types/block.go (hunk 1 above)
Comment (chaosma) [reply core/types/block.go]: removed for extV3

File: internal/params/config.go (hunk 2 above)
Comment (chaosma) [reply internal/params/config.go]: removed

**Pull request merged**